syntax = "proto3";
package assistant;

import "google/protobuf/empty.proto";

enum AuthMethod {
    QR_CODE = 0;
    PHONE = 1;
}

message MessageResponse {
  uint64 id = 1;
  int64 chat_id = 2;
//  bool can_be_forwarded = 3;
}

message ForwardMessagesRequest {
  int64 tg_user_id = 1;
  int64 from_chat_id = 2;
  int64 chat_id = 3;
  repeated uint64 message_ids = 4;

  optional bool disable_notification = 5;
}

message SendVideoRequest {
  int64 tg_user_id = 1;
  int64 chat_id = 2;
  string video_path = 3;

  optional string caption = 4;
  optional bool disable_notification = 5;
}

message SendFilesRequest {
  int64 tg_user_id = 1;
  int64 chat_id = 2;
  repeated string files = 3;

  optional string caption = 4;
  optional bool disable_notification = 5;
}

message SendTextRequest {
  int64 tg_user_id = 1;
  int64 chat_id = 2;
  string text = 3;
  bool disable_notification = 4;
}

message BooleanValue {
  bool value = 1;
}

message SendAuthorizationRequest {
  int64 tg_user_id = 1;
  AuthMethod auth_method = 2;
}

message SendCheckAuthorizationRequest {
  int64 tg_user_id = 1;
}

message SendLogoutRequest {
  int64 tg_user_id = 1;
}

service Assistant {
  rpc send_video(SendVideoRequest) returns (MessageResponse);
  rpc send_files(SendFilesRequest) returns (MessageResponse);
  rpc send_text(SendTextRequest) returns (MessageResponse);
  rpc forward_messages(ForwardMessagesRequest) returns (google.protobuf.Empty);
  rpc is_authorized(SendCheckAuthorizationRequest) returns (BooleanValue);
  rpc authorize_user(SendAuthorizationRequest) returns (google.protobuf.Empty);
  rpc logout_user(SendLogoutRequest) returns (google.protobuf.Empty);
}
